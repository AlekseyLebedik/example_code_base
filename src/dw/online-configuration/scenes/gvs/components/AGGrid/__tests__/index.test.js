import React from 'react';

import { shallow } from 'enzyme';

import GVSAGGrid from '..';

jest.mock('react-redux');
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams() {
    return { titleId: 1 };
  },
}));
const FakeRenderer = () => null;
const COLUMN_DEFS = [{ field: 'name' }, { field: 'value' }];

describe('GVSAGGrid', () => {
  it('adds sidebar to the grid options', () => {
    const wrapper = shallow(
      <GVSAGGrid
        columnDefs={COLUMN_DEFS}
        components={{ fakeRenderer: FakeRenderer }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
