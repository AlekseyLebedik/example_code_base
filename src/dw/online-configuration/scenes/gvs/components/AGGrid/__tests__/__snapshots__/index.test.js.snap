// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GVSAGGrid adds sidebar to the grid options 1`] = `
<AGGrid
  async={false}
  asyncGridProps={Object {}}
  autoSizeColumns={true}
  autoSizeMaxWidth={2000}
  columnDefs={
    Array [
      Object {
        "field": "name",
      },
      Object {
        "field": "value",
      },
    ]
  }
  context={Object {}}
  gridOptions={
    Object {
      "alignRowCenter": true,
      "components": Object {
        "fakeRenderer": [Function],
      },
      "sideBar": Object {
        "toolPanels": Array [
          Object {
            "iconKey": "columns",
            "id": "columns",
            "labelDefault": "Columns",
            "labelKey": "columns",
            "toolPanel": "agColumnsToolPanel",
            "toolPanelParams": Object {
              "suppressColumnExpandAll": true,
              "suppressColumnFilter": true,
              "suppressColumnSelectAll": true,
              "suppressPivotMode": true,
              "suppressPivots": true,
              "suppressRowGroups": true,
              "suppressValues": true,
            },
          },
        ],
      },
    }
  }
  onCellClicked={[Function]}
  onColumnMoved={[Function]}
  onColumnResized={[Function]}
  onColumnVisible={[Function]}
  onFirstDataRendered={[Function]}
  onGridReady={[Function]}
  onGridSizeChanged={[Function]}
  onRowSelected={[Function]}
  rowData={Array []}
/>
`;
